src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java

```diff
@@ -446,46 +465,27 @@ private void processCreatedFiles(String siteId, String creator,
 		Site site = sitesServiceInternal.getSite(siteId);
 		User userObj = userServiceInternal.getUserByGitName(creator);
 
-		StudioDBScriptRunner studioDBScriptRunner = studioDBScriptRunnerFactory.getDBScriptRunner();
-
-		Path createdFileScriptPath = null;
-		// TODO: SJ: Refactor to avoid string literals
-		try {
-			Path studioTempDir = getStudioTemporaryFilesRoot();
-			String createdFileScriptFilename = CREATED_FILES_SCRIPT_PREFIX + UUID.randomUUID();
-			createdFileScriptPath = Files.createTempFile(studioTempDir, createdFileScriptFilename, SQL_SCRIPT_SUFFIX);
-
-			// Variable in lambda needs to be final
-			final Path scriptPath = createdFileScriptPath;
-
-			MutableLong itemCount = new MutableLong(0);
-			ThrowingRunnable checkCounter = getCheckCounterFunction(itemCount, siteId, studioDBScriptRunner, scriptPath);
+		MutableLong itemCount = new MutableLong(0);
+		try (SqlSession sqlSession = sqlSessionFactory.openSession(ExecutorType.BATCH)) {
+			ThrowingRunnable checkCounter = getCheckCounterFunction(sqlSession, itemCount, siteId);
 			contentRepositoryV2.forAllSitePaths(siteId,
 				directory -> {
-					processCreatedDirectory(scriptPath, site.getId(), directory, userObj.getId(), now);
+					processCreatedDirectory(sqlSession, site.getSiteId(), directory, userObj.getId(), now);
```

Comment:
Can we get the mapper out of the session and pass it here instead of the actual sqlSession object? That would create a single mapper for all updates.

Reply:
I updated to send `dependencyDao` and `itemDao` from the caller. 
Please note that we cannot use the initialized DAO beans as it will update DB immediately instead of using the session.